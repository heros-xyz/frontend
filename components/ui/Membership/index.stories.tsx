import type { Meta, StoryObj } from "@storybook/react";
import Membership from "./index";

const meta: Meta<typeof Membership> = {
  title: "Athlete/Membership",
  component: Membership,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
  argTypes: {
    buttonContent: {
      if: { arg: "isMembership", truthy: false }
    },
    tier: {
      if: { arg: "isMembership"}
    },
  },
};

export default meta;
type Story = StoryObj<typeof Membership>;

export const MembershipTemplate: Story = {
  args: {
    title: "Membership",
    tier: 1,
    isMembership: true,
    buttonContent: "Add tier"
  },
};
