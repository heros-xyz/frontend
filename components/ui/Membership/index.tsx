import { Box, Button, Flex, Heading, Skeleton, Text } from "@chakra-ui/react";
import React from "react";

interface IWalletProps {
  title: string;
  tier?: number;
  isMembership: boolean;
  buttonContent?: string;
  onClickManage: () => void;
  onAddTier?: () => void;
  isLoading: boolean;
}

const Membership: React.FC<IWalletProps> = ({
  title,
  tier,
  isMembership,
  buttonContent,
  onClickManage,
  onAddTier,
  isLoading,
}) => {
  return (
    <Box
      bg="acccent.1"
      borderRadius={{ base: "lg", xl: "xl" }}
      px={{ base: 5, xl: 7 }}
      py={{ base: 3, xl: 5 }}
    >
      <Flex mb="1.5" justify="space-between" align="center">
        <Heading
          color="primary"
          fontSize={{ base: "md", xl: "xl" }}
          fontWeight="extrabold"
          textTransform="capitalize"
          lineHeight="140%"
        >
          {title}
        </Heading>
        {isMembership && (
          <Text
            cursor="pointer"
            fontSize={{ base: "xs", xl: "lg" }}
            fontWeight="medium"
            color="acccent.2"
            lineHeight="140%"
            onClick={onClickManage}
          >
            Manage
          </Text>
        )}
      </Flex>
      <Text
        fontSize={{ base: "xs", xl: "md" }}
        fontWeight="normal"
        color="primary"
      >
        {isMembership
          ? "You are currently offering."
          : "You are not offering any membership."}
      </Text>
      {isMembership ? (
        <Text
          mt={{ base: 3.5, xl: 7 }}
          fontSize={{ base: "md", xl: "xl" }}
          fontWeight="medium"
          color="primary"
          lineHeight="140%"
        >
          <Skeleton isLoaded={!isLoading} as="span">
            {tier}
          </Skeleton>{" "}
          Tier
        </Text>
      ) : (
        <Flex justify={{ base: "center", xl: "flex-end" }}>
          <Button
            variant="primary"
            width={{ base: "100%", xl: "auto" }}
            fontSize={{ base: "md", xl: "xl" }}
            h="48px"
            mt={{ base: 5, xl: 3.5 }}
            onClick={onAddTier}
          >
            {buttonContent}
          </Button>
        </Flex>
      )}
    </Box>
  );
};

export default Membership;
