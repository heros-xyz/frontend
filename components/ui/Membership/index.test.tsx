import { render } from "@testing-library/react";
import Membership from "./index";
it("renders membership unchanged", () => {
  const { container } = render(
    <Membership title="Membership" tier={1} isMembership={true} />
  );
  expect(container).toMatchSnapshot();
});
