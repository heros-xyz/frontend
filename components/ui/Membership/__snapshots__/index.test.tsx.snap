// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders membership unchanged 1`] = `
<div>
  <div
    class="css-1h0mgil"
  >
    <div
      class="css-5tnwgz"
    >
      <h2
        class="chakra-heading css-ayg11n"
      >
        Membership
      </h2>
      <p
        class="chakra-text css-ohbwvd"
      >
        Manage
      </p>
    </div>
    <p
      class="chakra-text css-71uet5"
    >
      You are currently offering.
    </p>
    <p
      class="chakra-text css-zvd5jj"
    >
      <span
        class="chakra-skeleton css-cdkrf0"
      >
        1
      </span>
       
      tier
    </p>
  </div>
</div>
`;
